### PR TITLE
use the "X-Forwarded-For" for whatIsMyIp

### PR DESCRIPTION
**Pull request by eishay**

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/mobile/MobileAuthController.scala [hunk 1]
```diff
@@ -38,7 +38,10 @@ class MobileAuthController @Inject() (
 
   private implicit val readsOAuth2Info = Json.reads[OAuth2Info]
 
-  def whatIsMyIp() = Action { request => Ok(request.remoteAddress) }
+  def whatIsMyIp() = Action { request =>
+    val ip = request.headers.get("X-Forwarded-For").getOrElse(request.remoteAddress)
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/mobile/MobileAuthController.scala:42]: Should we even fall back to `request.remoteAddress`? It's always useless / misleading, right?

Comment (eishay) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/mobile/MobileAuthController.scala:42]: its actually working perfectly right afaik. We have only `request.remoteAddress` in prod at this moment and it returns the right ip:
https://api.kifi.com/m/1/whatismyip
I checked it with both desktop and mobile browsers.
Adding `X-Forwarded-For` just for extra assurance.

Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/mobile/MobileAuthController.scala:42]: Oh nice. Nothing to see here :)

On April 16, 2014 at 11:37:33 AM, Eishay Smith (notifications@github.com) wrote:

In kifi-backend/modules/shoebox/app/com/keepit/controllers/mobile/MobileAuthController.scala:

Comment (eishay) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/mobile/MobileAuthController.scala:42]: nice

Comment (eishay) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/mobile/MobileAuthController.scala:42]: second test

Comment (2is10) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/mobile/MobileAuthController.scala:42]: Here is a good example of how _not_ to implement reply-by-email.

Comment (eishay) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/mobile/MobileAuthController.scala:42]: yes, that's what we're testing. they're sort of doing a better job with
emails coming from gmail then Andrew's Airmail Mac client.

Comment (2is10) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/mobile/MobileAuthController.scala:42]: If you visit this thread on github.com, it looks like they’re actually
doing a reasonable job of detecting quoted text. The problem is just that
they’re re-sending it with loss of fidelity.

In our case, I think it makes sense to aggressively eliminate quoted text
and auto-generated reply headings.

**Pull request merged**